datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -106,6 +106,55 @@ pub fn binary_operator_data_type(
     }
 }
 
+pub fn get_result_type(
```

Comment:
I think we need some docstrings on this function. Something like:

```suggestion
/// returns the resulting type of a binary expression evaluating the `op` with the left and right hand types 
pub fn get_result_type(
```

Also calling this method `result_type` rather than `get_result_type` might be more consistent with the conventions in the codebase (this is a minor point)